docs/examples/custom/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	examples "github.com/cdevents/sdk-go/docs/examples"
+	cdevents "github.com/cdevents/sdk-go/pkg/api"
+	cdeventsv04 "github.com/cdevents/sdk-go/pkg/api/v04"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+const customSchemaUri = "https://myregistry.dev/schemas/cdevents/quota-exceeded/0_1_0"
+
+type Quota struct {
+	User      string `json:"user"`      // The use the quota applies ot
+	Limit     string `json:"limit"`     // The limit enforced by the quota e.g. 100Gb
+	Current   int    `json:"current"`   // The current % of the quota used e.g. 90%
+	Threshold int    `json:"threshold"` // The threshold for warning event e.g. 85%
+	Level     string `json:"level"`     // INFO: <threshold, WARNING: >threshold, <quota, CRITICAL: >quota
+}
+
+func main() {
+	var ce *cloudevents.Event
+	var c cloudevents.Client
+
+	// Define the event type
+	eventType := cdevents.CDEventType{
+		Subject:   "quota",
+		Predicate: "exceeded",
+		Version:   "0.1.0",
+		Custom:    "myregistry",
+	}
+
+	// Define the content
+	quotaRule123 := Quota{
+		User:      "heavy_user",
+		Limit:     "50Tb",
+		Current:   90,
+		Threshold: 85,
+		Level:     "WARNING",
+	}
+
+	// Create the base event
+	event, err := cdeventsv04.NewCustomTypeEvent()
+	examples.PanicOnError(err, "could not create a cdevent")
+	event.SetEventType(eventType)
+
+	// Set the required context fields
+	event.SetSubjectId("quotaRule123")
+	event.SetSource("my/first/cdevent/program")
+
+	// Set the required subject fields
+	event.SetSubjectContent(quotaRule123)
```

Comment:
I was debating between explicit setters/getters and providing some kind of context object where users could stash fields to be set:
`event.WithSubjectId("foo").WithSource("bar")` 

But that leads to as many `WithX` methods as `SetX` methods to be defined, and I didn't really see an advantage. For the subject content, I introduced `SetSubjectContent` which allows setting the whole content from a Struct directly rather than field by field. 

I'd like to better understand what you are proposing here, maybe something for the next WG.